src/services/user/hooks/handle-facebook-create.js

```diff
@@ -0,0 +1,52 @@
+
+const handleFacebookCreate = () => (hook) => {
+  // Pass through this hook if there's no facebook data
+  if (!hook.data.facebookId
+    || !hook.data.facebook
+    || !hook.data.facebook.email
+    || (!hook.data.facebook.name
+      && !hook.data.facebook.first_name
+      && !hook.data.facebook.last_name)
+    ) {
+    return hook;
+  }
+
+  const userService = hook.app.service('/users');
```

Comment:
Yes. It will call the *create* service again. However, the 2nd call will fail the guard check of this before hook (it will be called with data composing of facebookId, email, name fields only) and will pass through to the normal create.